File: java/src/test/java/problems1_100/LongestPalindromicSubstringTest.java
package problems1_100;

import org.junit.Test;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class LongestPalindromicSubstringTest extends SampledTest {
    @Test
    public void test() {
        for (LongestPalindromicSubstring solution : LongestPalindromicSubstring.values()) {
            for (LongestPalindromicSubstringSample sample
                : getSamples(LongestPalindromicSubstringSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.longestPalindrome(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/LongestPalindromicSubstringTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class LongestPalindromicSubstringTest : SampledTest() {
    @Test
    fun test() =
        LongestPalindromicSubstring.entries.forEach { solution ->
            getSamples<Array<LongestPalindromicSubstringSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.longestPalindrome(input.s))
                        .isEqualTo(output)
                }
        }
}
